traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/router/TrafficRouter.java

```diff
@@ -675,17 +678,40 @@ public CacheLocation getCoverageZoneCacheLocation(final String ip, final String
 			return cacheLocation;
 		}
 
-		// We had a hit in the CZF but the name does not match a known cache location.
-		// Check whether the CZF entry has a geolocation and use it if so.
-		return getClosestCacheLocation(cacheRegister.filterAvailableLocations(deliveryServiceId), networkNode.getGeolocation(), cacheRegister.getDeliveryService(deliveryServiceId));
+		if (networkNode.getBackupLoc() != null ) {
+			for (final String element : networkNode.getBackupLoc()) {
+				final CacheLocation bkCacheLocation = getCacheRegister().getCacheLocationById(element);
+				if (bkCacheLocation != null && !getSupportingCaches(bkCacheLocation.getCaches(), deliveryService).isEmpty()) {
+					LOGGER.info("Got backup CZ cache group " + bkCacheLocation.getId() + " for " + ip + ", ds " + deliveryServiceId);
+					if (track != null) {
+						track.setFromBackupCzGroup(true);
+					}
+					return bkCacheLocation;
+				}
+			}
+		} 
+
+		if ((networkNode.getBackupLoc() != null && networkNode.isUseClosest()) || (networkNode.getBackupLoc() == null)) {
```

Comment:
Even if the backupZones list is empty and the operator sets useClosest=true, I think we should enter this loop because its equivalent to trying to route to the backupZones when they are all down/unavailable.

Reply:
I think , this condition in NetworkNode.java will take care of this, where we parse and store the useClosest flag.
if (backupConfigNode != null && backupConfigNode.has("list")) { 
...
}

Reply:
With this the following config will do the closest group if backup list is empty
 "backupZones":{ 
      "list": [],
      "fallbackToClosestGroup": true 
   },

and wont do closest group with the following:
 "backupZones":{ 
      "list": [],
      "fallbackToClosestGroup": false 
   },

Reply:
ok, i'm trying to simplify this conditional, because its awkward to have opposite checks on both sides of an OR. Do we really need to check for `networkNode.getBackupLoc()` here?

Reply:
Got you. Let me fix this.